packages/app-desktop/gui/NoteEditor/utils/contextMenu.ts

```diff
@@ -100,17 +103,28 @@ export function menuItems(dispatch: Function): ContextMenuItems {
 			onAction: async (options: ContextMenuOptions) => {
 				await openItemById(options.resourceId, dispatch);
 			},
-			isActive: (itemType: ContextMenuItemType) => itemType === ContextMenuItemType.Image || itemType === ContextMenuItemType.Resource,
+			isActive: (itemType: ContextMenuItemType, options: ContextMenuOptions) => itemType === ContextMenuItemType.Image || (itemType === ContextMenuItemType.Resource && options.resourceId),
 		},
 		saveAs: {
 			label: _('Save as...'),
 			onAction: async (options: ContextMenuOptions) => {
-				const { resourcePath, resource } = await resourceInfo(options);
-				const filePath = await bridge().showSaveDialog({
-					defaultPath: resource.filename ? resource.filename : resource.title,
-				});
+				let defaultPath = '';
+				let resourcePath = null;
+				let resourceData = null;
```

Comment:
One of the reason your implementation is confusing is because of the use of non-const variable. You mutate data and it means different things depending on some remote context. Instead please use const everywhere and name each variable carefully. In that same block I see things like "resourceContent", "resourceData", etc. that's the same meaning, yet it's used for different things.  No-one will understand this code 2 months from now (even now, I can't make any sense of it).